src/oracles/ChainlinkPriceOracle.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0 <0.9.0;
+
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+
+/**
+ * @title Chainlink Price Oracle
+ * @author GUNBOATs
+ * @dev This contract creates an oracle that is compatible with the IPriceOracle
+ * interface and can be used by a CoW AMM to determine the current price of the
+ * traded tokens from Chainlink Data Feeds with decimals lower than 18.
+ */
+contract ChainlinkPriceOracle is IPriceOracle {
+    /**
+    * @param token0Feed Address of token0 oracle
+    * @param token1Feed Address of token1 oracle
+    * @param timeThreshold amount of seconds before it consider as stale oracle
+    */
+    struct Data {
+        address token0Feed;
+        address token1Feed;
+        uint256 timeThreshold;
+    }
+    /**
+     * @dev There is no mapping between Chainlink oracle and the token address,
+     * so the user must check the that the oracle is conresponded to the token by themself.
+     * @inheritdoc IPriceOracle
+     */
+    function getPrice(address, address, bytes calldata data)
+        external
+        view
+        returns (uint256 priceNumerator, uint256 priceDenominator)
+    {
+        Data memory OracleData = abi.decode(data, (Data));
+        AggregatorV3Interface token0Feed = AggregatorV3Interface(OracleData.token0Feed);
+        AggregatorV3Interface token1Feed = AggregatorV3Interface(OracleData.token1Feed);
+        ( /* uint80 roundId*/
+            , int256 token0Answer, /* uint256 startedAt */, uint256 token0Timestamp, /* uint80 answerInRound */
+        ) = token0Feed.latestRoundData();
+        ( /* uint80 roundId*/
+            , int256 token1Answer, /* uint256 startedAt */, uint256 token1Timestamp, /* uint80 answerInRound */
+        ) = token1Feed.latestRoundData();
+        uint256 timestamp = block.timestamp;
+        require(
+            timestamp - token0Timestamp < OracleData.timeThreshold 
+            && timestamp - token1Timestamp < OracleData.timeThreshold, 
+            "stale oracle"
+            );
```

Comment:
`getPrice` should always return a proper revert message, otherwise as this is in the code path used by `getTradeableOrderWithSignature` that is polled by the watch-tower, if any reversions do not adhere to strict requirements, the watch-tower will drop the monitoring of the conditional order altogether. See [custom error codes](https://docs.cow.fi/cow-protocol/reference/contracts/periphery/composable-cow#custom-error-codes) in the documentation for more information on this.

I would suggest likely reverting with `PollTryAtEpoch(uint256 timestamp, string)` asking the watch-tower to try again at some `block.timestamp + backoff` with the string being set to `stale oracle`.

Reply:
Is it ok to hardcode backoff duration?

Reply:
I think maybe this would be good to pass in via the price oracle data as the staleness configuration may be somewhat useful to configure by specific chainlink oracle.